### PR TITLE
[Refactor] Read tablet schema from metadata in LakeDataSource

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.2

File: be/src/connector/lake_connector.cpp [hunk 1]
```diff
@@ -311,9 +310,11 @@ Status LakeDataSource::get_next(RuntimeState* state, ChunkPtr* chunk) {
 }
 
 Status LakeDataSource::get_tablet(const TInternalScanRange& scan_range) {
-    _version = strtoul(scan_range.version.c_str(), nullptr, 10);
-    ASSIGN_OR_RETURN(auto tablet, ExecEnv::GetInstance()->lake_tablet_manager()->get_tablet(scan_range.tablet_id));
-    ASSIGN_OR_RETURN(_tablet_schema, tablet.get_schema());
+    int64_t tablet_id = scan_range.tablet_id;
+    int64_t version = strtoul(scan_range.version.c_str(), nullptr, 10);
+    ASSIGN_OR_RETURN(_tablet, ExecEnv::GetInstance()->lake_tablet_manager()->get_tablet(tablet_id, version));
+    CHECK(_tablet.valid());
```
Comment (kevincai) [inline be/src/connector/lake_connector.cpp:316]: maybe DCHECK instead of check? avoid crash BE in release mode?

Comment (sduzh) [reply be/src/connector/lake_connector.cpp:316]: Change CHECK to DCHECK will not prevent process crash in release mode, but just let it crash in the nextl ine..

Comment (sduzh) [reply be/src/connector/lake_connector.cpp]: I removed this assertition

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**